src/interactions/panZoomInteraction.ts

```diff
@@ -3,14 +3,74 @@ namespace Plottable.Interactions {
   export type PanCallback = () => void;
   export type ZoomCallback = () => void;
 
+  /**
+   * Performs a zoom transformation of the `value` argument scaled by the
+   * `zoom` argument about the point defined by the `center` argument.
+   */
+  export function zoomAt(value: number, zoom: number, center: number) {
+    return center - (center - value) * zoom;
+  }
+
+  /**
+   * Type guarded function to check if the scale implements the
+   * `TransformableScale` interface. Unfortunately, there is no way to do
+   * runtime interface typechecking, so we have to explicitly list all classes
+   * that implement the interface.
+   */
+  export function isTransformable(scale: any): scale is TransformableScale {
+    return (scale instanceof Plottable.QuantitativeScale ||
+      scale instanceof Plottable.Scales.Category);
+  }
+
+  /**
+   * This interface abstracts the necessary API for implementing pan/zoom on
+   * various types of scales.
+   *
+   * Due to some limitations of d3, for example category scales can't invert
+   * screen space coordinates, we nevertheless allow the scale types to support
+   * pan/zoom if they implement this interface. See `Category`'s
+   * `_d3TransformationScale` for more info.
+   */
+  export interface TransformableScale {
```

Comment:
Ohhh my god Github hid this whole file from me... Okay this makes a lot more sense. Imo this interface belongs in `Plottable.Scales` instead of `Plottable.Interactions` (and you can `type TransformableScale = Plottable.Scales.TransformableScale` in this file to prevent needing to type it out everywhere)